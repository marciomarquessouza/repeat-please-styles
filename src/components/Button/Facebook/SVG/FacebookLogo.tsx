import React from 'react';
import Svg, { Path } from 'react-native-svg';
import { View, ViewStyle } from 'react-native';

export interface IFacebookLogoProps {
	style?: ViewStyle;
}

export const FacebookLogo = (props: IFacebookLogoProps): JSX.Element => (
	<View {...props} data-test="FacebookLogo">
		<Svg width={15} height={28} viewBox="0 0 15 28" fill="none">
			<Path
				d="M10.039 27.893v-12.31h4.132l.618-4.796h-4.75V7.724c0-1.39.386-2.336 2.377-2.336h2.54V1.096c-.439-.059-1.947-.19-3.701-.19-3.663 0-6.17 2.236-6.17 6.342v3.538H.943v4.797h4.142v12.31h4.954z"
				fill="#fff"
			/>
		</Svg>
	</View>
);
