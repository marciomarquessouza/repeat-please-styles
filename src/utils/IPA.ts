export const IPA = {
	ʌ: {
		symbol: 'ʌ',
		type: 'vowel',
		examples: ['cUp', 'lUck'],
	},
	'ɑ:': {
		symbol: 'ɑ:',
		type: 'vowel',
		examples: ['Arm', 'fAther'],
	},
	æ: {
		symbol: 'æ',
		type: 'vowel',
		examples: ['cAt', 'blAck'],
	},
	e: {
		symbol: 'e',
		type: 'vowel',
		examples: ['mEt', 'bEd'],
	},
	ə: {
		symbol: 'ə',
		type: 'vowel',
		examples: ['Away', 'cinemA'],
	},
	'ɜ:': {
		symbol: 'ɜ:',
		type: 'vowel',
		examples: ['tURn', 'lEARn'],
	},
	ɪ: {
		symbol: 'ɪ',
		type: 'vowel',
		examples: ['hIt', 'sITTING'],
	},
	'i:': {
		symbol: 'i:',
		type: 'vowel',
		examples: ['SEE', 'HEAT'],
	},
	ɒ: {
		symbol: 'ɒ',
		type: 'vowel',
		examples: ['HOT', 'ROCK'],
	},
	'ɔ:': {
		symbol: 'ɔ:',
		type: 'vowel',
		examples: ['CALL', 'FOUR'],
	},
	ʊ: {
		symbol: 'ʊ',
		type: 'vowel',
		examples: ['PUT', 'COULD'],
	},
	'u:': {
		symbol: 'u:',
		type: 'vowel',
		examples: ['BLUE', 'FOOD'],
	},
	aɪ: {
		symbol: 'aɪ',
		type: 'vowel',
		examples: ['FIVE', 'EYE'],
	},
	aʊ: {
		symbol: 'aʊ',
		type: 'vowel',
		examples: ['NOW', 'OUT'],
	},
	eɪ: {
		symbol: 'eɪ',
		type: 'vowel',
		examples: ['SAY', 'EIGHT'],
	},
	oʊ: {
		symbol: 'oʊ',
		type: 'vowel',
		examples: ['GO', 'HOME'],
	},
	ɔɪ: {
		symbol: 'ɔɪ',
		type: 'vowel',
		examples: ['BOY', 'JOIN'],
	},
	eə: {
		symbol: 'eə',
		type: 'vowel',
		examples: ['WHERE', 'AIR'],
	},
	ɪə: {
		symbol: 'ɪə',
		type: 'vowel',
		examples: ['NEAR', 'HERE'],
	},
	ʊə: {
		symbol: 'ʊə',
		type: 'vowel',
		examples: ['PURE', 'TOURIST'],
	},
};
