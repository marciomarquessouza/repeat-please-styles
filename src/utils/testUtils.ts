import { ShallowWrapper } from 'enzyme';

export const findByDataTest = (
	wrapper: ShallowWrapper,
	dataTest: string,
): ShallowWrapper => {
	return wrapper.find(`[data-test="${dataTest}"]`);
};
