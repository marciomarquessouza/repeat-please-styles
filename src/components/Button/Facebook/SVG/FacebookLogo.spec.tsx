import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { FacebookLogo, IFacebookLogoProps } from './FacebookLogo';
import { findByDataTest } from '../../../../utils/testUtils';

const setup = (props: IFacebookLogoProps): ShallowWrapper => {
	return shallow(<FacebookLogo {...props} />);
};

describe('Facebook Logo Basic Test', () => {
	it('should render Facebook Logo Component properly', () => {
		const wrapper = setup({});
		const facebookLogo = findByDataTest(wrapper, 'FacebookLogo');
		expect(facebookLogo).toHaveLength(1);
		expect(facebookLogo).toMatchSnapshot();
	});
});
