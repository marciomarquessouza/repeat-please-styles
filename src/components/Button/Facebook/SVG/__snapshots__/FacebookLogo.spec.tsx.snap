// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facebook Logo Basic Test should render Facebook Logo Component properly 1`] = `ShallowWrapper {}`;
